src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java

```diff
@@ -86,14 +87,13 @@ public void run(ActionContext context) throws Exception {
 
     CryptoKeyName cmekKeyName = CmekUtils.getCmekKey(config.cmekKey, context.getArguments().asMap(), collector);
     collector.getOrThrowException();
-    if (cmekKeyName != null) {
-      builder.setDestinationEncryptionConfiguration(
-        EncryptionConfiguration.newBuilder().setKmsKeyName(cmekKeyName.toString()).build());
-    }
 
     // Save the results of the query to a permanent table.
-    if (config.getDataset() != null && config.getTable() != null) {
-      builder.setDestinationTable(TableId.of(config.getDataset(), config.getTable()));
+    String datasetName = config.getDataset();
+    String tableName = config.getTable();
+    String datasetProjectId = config.getDatasetProject();
+    if (datasetProjectId != null && datasetName != null && tableName != null) {
+      builder.setDestinationTable(TableId.of(datasetProjectId, datasetName, tableName));
```

Comment:
should this be guarded by getStoreResults too ?

Reply:
ohh, yeah missed it